### PR TITLE
epic: integration tests (v5+)

### PR DESCRIPTION
**Pull request by ala-n**
This is just an early draft of setting up visual screenshot testing.

In the scope of that epic created inner submodule with:
  - `jest-puppeteer` environment 
  - `jest-image-snapshot` image matching tool (TODO: to consider faster alternatives if required)
  - An environment uses TypeScript and global test linting rules
  - The puppeteer environment bootstrapped to launch tests in headless (new version) of Chrome with the local server auto start on port 3005 (TODO: to change in case of conflicts)
  - Added Gherkin spec declaration support
  
There is a sample test created to check the homepage screen of the upper area on SM screen breakpoint
TODO: create more relevant test samples

### CODE REVIEW

File: e2e/reporters/reporter.js [hunk 1]
```diff
@@ -0,0 +1,94 @@
...
+  buildTestStat(test, testPath) {
+    const { ancestorTitles, status, title, duration} = test;
+    const filename = path.basename(testPath);
+    const name = ancestorTitles.join(' > ');
+    const statBase = {name, filename, status, title, time: duration};
+
+    if (status === 'passed') return statBase;
+
+    const snapshotParts = [filename, ...ancestorTitles, title, '1-snap', 'diff'];
+    const snapshotName = snapshotParts.map(sanitize).join('-') + '.png'
```
Comment (yadamskaya) [inline e2e/reporters/reporter.js:55]:
```suggestion
    const snapshotName = snapshotParts.map(sanitize).join('-') + '.png';
```

File: e2e/setup/scenarios.screenshot.ts [hunk 2]
```diff
@@ -0,0 +1,27 @@
+import {cucumber} from '../transformer/gherkin';
+import type {TestEnv} from './scenarious.world';
+import type {MatchImageSnapshotOptions} from 'jest-image-snapshot';
+
+const DIFF_CONFIG: MatchImageSnapshotOptions = {
+  customDiffDir: '.diff',
```
Comment (yadamskaya) [inline e2e/setup/scenarios.screenshot.ts:6]: minor
```suggestion
  customDiffDir: '.diff'
```

**Review by yadamskaya — APPROVED** [review]

**Review by julia-murashko — APPROVED** [review]

File: e2e/reporters/reporter.js [hunk 3]
```diff
@@ -0,0 +1,94 @@
...
+      numTotalTestSuites: results.numTotalTestSuites,
+      startTime: results.startTime,
+      startTimeStr,
+      totalTime,
+      totalTimeStr
+    };
+  }
+
+  buildTestStat(test, testPath) {
+    const { ancestorTitles, status, title, duration} = test;
```
Comment (dshovchko) [inline e2e/reporters/reporter.js:47]:
```suggestion
    const {ancestorTitles, status, title, duration} = test;
```

File: e2e/README.md [hunk 4]
```diff
@@ -0,0 +1,106 @@
...
+It is designed to be used by the ESL team to test the ESL library by checking the visual consistency of `@exadel/esl-website` submodule.
+
+## Installation
+
+There are no extra pre-conditions to develop new visual tests locally, 
+everything to start will be already installed with root package installation.
+
+**⚠️ Except for one critical thing: all final snapshots should be created on Linux based OS (the latest version of Ubuntu is recommended) ⚠️**
+
+Visual testing depends on font rendering so snapshots created on different OS may be different.
```
Comment (fshovchko) [inline e2e/README.md:14]:
```suggestion
Visual testing depends on font rendering so snapshots created on different OS may vary.
```

File: e2e/README.md [hunk 5]
```diff
@@ -0,0 +1,106 @@
...
+## Introduction
+This is an internal package for visual automation testing the ESL. 
+It is designed to be used by the ESL team to test the ESL library by checking the visual consistency of `@exadel/esl-website` submodule.
+
+## Installation
+
+There are no extra pre-conditions to develop new visual tests locally, 
+everything to start will be already installed with root package installation.
+
+**⚠️ Except for one critical thing: all final snapshots should be created on Linux based OS (the latest version of Ubuntu is recommended) ⚠️**
```
Comment (ala-n) [inline e2e/README.md:12]:
```suggestion
**⚠️ Except for one critical thing: all final snapshots should be created on Linux based OS (the latest LTS version of Ubuntu is recommended) ⚠️**
```

File: e2e/README.md [hunk 6]
```diff
@@ -0,0 +1,106 @@
...
+   - `npm run test:e2e` (in the root package) or `npm run run` (in the sub-package) to run all tests (missing snapshots will be created automatically)
+   - `npm run test:e2e:update` (in the root package) or `npm run run:update` (in sub-package) to update all snapshots
+
+### Creating snapshots and running existing tests on Mac
+
+TODO: add instructions for Mac OS
+
+### Creating snapshots and running existing tests on Linux
+
+To create snapshots on Linux, you can use any Linux distribution, but the latest version of Ubuntu is recommended.
```
Comment (ala-n) [inline e2e/README.md:52]:
```suggestion
To create snapshots on Linux, you can use any Linux distribution, but the latest LTS version of Ubuntu is recommended.
```

**Review by dshovchko — APPROVED** [review]

File: e2e/reporters/printers.js [hunk 7]
```diff
@@ -0,0 +1,65 @@
...
+    text += `<table>\n`;
+    text += '<tr><th>Test</th><th>Status</th><th>Time</th></tr>\n';
+    for (const test of file.tests) {
+      const statusTest = test.status === 'passed' ? ':white_check_mark:' : ':x:';
+      const timeStr = test.time < 1000 ? `${test.time}ms` : `${test.time / 1000}s`;
+
+      text += `<tr><td>${test.name}:${test.title}</td><td>${statusTest}</td><td>${timeStr}</td></tr>\n`;
+
+      if (test.status !== 'passed' && test.hasSnapshot) {
+        text += `<tr><td colspan="3"><img src="${resolveURL(basePath, test.snapshot)}" alt="Test Diff ${test.snapshot}"/></td></tr>`;
```
[the remark this answers is not in the document]
Comment (ala-n) [reply e2e/reporters/printers.js:38]: loading='lazy' does not support in MD and definitely will be omitted by Github MD renderer (which is our target here)

[the remark this answers is not in the document]
Comment (dshovchko) [reply e2e/reporters/printers.js:38]: oh :(

**Review by fshovchko — APPROVED** [review]

**Pull request merged**